sdk/core/azure_core/benches/core_benchmarks.rs

```diff
@@ -0,0 +1,31 @@
+use azure_core::http::Url;
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use std::time::Duration;
+// how many params to add to the url
+const PARAMS: &[usize] = &[1, 10, 100, 1000];
+
+fn url_parsing_benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("url_parsing");
+    for num in PARAMS {
+        let mut url = String::new();
+        url.push_str("https://example.com/path/resource?param=value");
+
+        for i in 0..*num {
+            url.push_str(&format!("&param{}=value{}", i, i));
+        }
+        group.bench_function(BenchmarkId::new("parse_basic_url", num), |b| {
+            b.iter(|| {
+                let _url = Url::parse(&url).unwrap();
+            })
+        });
+    }
+    group.finish();
+}
+
+// Main benchmark group
+criterion_group! {
+    name = url_benches;
```

Comment:
I would rather see each `benchmarks.rs` that all crates should have focus on 1 or more hero scenarios, so let's not call the outer group scenario-specific. Maybe just the name of the file name (module) so `benchmarks` here. We'll use the package selection already on `cargo bench` as we do with other commands for individual benchmark selection.

Reply:
done